helper/schema/schema.go

```diff
@@ -948,6 +954,11 @@ func checkKeysAgainstSchemaFlags(k string, keys []string, topSchemaMap schemaMap
 			return fmt.Errorf("%s cannot contain Computed(When) attribute (%s)", k, key)
 		}
 	}
+
+	if requireSelfReference && !foundSelfReference {
```

Comment:
`AtLeastOneOf` and `ExactlyOneOf` were implemented with an idempotent line that adds itself and removes duplicate, it allows the self reference to be optional. I would change this PR to have a flag for `allowSelfRef` and ditch the requirement for it, but rather disallow for `ConflictsWith` (and possibly `RequiredWith`)

Reply:
Unless we remove that line in `validateRequiredWith` I would actually ONLY have this error for `ConflictsWith`, just to be consistent, I will approve either final implementation.